### DISCUSSION TITLE
services higress-gateway pending

### DISCUSSION THREAD

**Discussion by zskeep** [alibaba/higress#297] [General]
k8s: v1.22.12
OS:Ubuntu 22.04.2 LTS
helm:v3.8.1

helm install

code:
helm repo add istio https://istio-release.storage.googleapis.com/charts
helm install istio-base istio/base -n istio-system

helm upgrade higress -n higress-system --set global.enableIstioAPI=true higress.io/higress --set global.enableStatus=true

discussions：
I cannot access the height management page outside the cluster
svc - higress-gateway is pending

kubectl get svc -n higress-system 
NAME                 TYPE           CLUSTER-IP      EXTERNAL-IP   PORT(S)                                                    AGE
higress-console      ClusterIP      10.101.39.254   <none>        8080/TCP                                                   109m
higress-controller   ClusterIP      10.98.6.244     <none>        8888/TCP,15051/TCP,15010/TCP,15012/TCP,443/TCP,15014/TCP   109m
higress-gateway      LoadBalancer   10.99.129.15    <pending>     80:31077/TCP,443:31838/TCP                                 109m

**Comment by CH3CHO** [comment]
Are you installing Higress to a local cluster or not? Could you try to run `kubectl describe pod` on the pending higress-gateway pod and see what's wrong there?

**Reply by zskeep** [reply]
kubectl get pod -n istio-system 
NAME                      READY   STATUS    RESTARTS   AGE
istiod-56f5f79c79-sbf45   1/1     Running   0          72

kubectl get pod -n higress-system 
NAME                                  READY   STATUS    RESTARTS   AGE
higress-console-59cd96df6c-vjv5z      1/1     Running   0          60m
higress-controller-694dd857cd-jpw46   2/2     Running   0          59m
higress-gateway-f98c484bd-bcpff       1/1     Running   0          14s

kubectl describe pod -n higress-system higress-gateway-f98c484bd-bcpff
Events:
  Type    Reason     Age   From               Message
  ----    ------     ----  ----               -------
  Normal  Scheduled  48s   default-scheduler  Successfully assigned higress-system/higress-gateway-f98c484bd-bcpff to k8s03
  Normal  Pulled     47s   kubelet            Container image "higress-registry.cn-hangzhou.cr.aliyuncs.com/higress/gateway:1.0.0-rc" already present on machine
  Normal  Created    47s   kubelet            Created container higress-gateway
  Normal  Started    47s   kubelet            Started container higress-gateway

**Reply by CH3CHO** [reply]
It looks all right here. How did you access the gateway outside the cluster? Does `http://10.99.129.15/` return a 404 error?

**Reply by zskeep** [reply]
I have no idea for gateway, and cannot open the website.
So,I need a gateway？Which one is more recommended,LB/ingress-nginx/HostNetwork/NodePort,or other one.

**Reply by CH3CHO** [reply]
1. Are you installing Higress to a local cluster or a standard production K8s cluster?
2. You can try following steps:
   1. Execute `sudo kubectl port-forward service/higress-gateway -n higress-system 80:80`.
      If the command fails, run this one instead: `kubectl port-forward service/higress-gateway -n higress-system 54321:80`.
   2. Add a new line into `/etc/hosts`: `127.0.0.1 console.higress.io`
   3. Open `http://console.higress.io/` in your browser. If you use the alternative command in step 1, change the URL to `http://console.higress.io:54321/`.
   4. If anything goes wrong, please provide corresponding screenshots so we can know some details.

**Reply by zskeep** [reply]
How to configure the ingress class(name:higress)? Perhaps, could you provide a configuration template.

**Reply by CH3CHO** [reply]
You can check out this: https://github.com/alibaba/higress/blob/main/samples/hello-world/quickstart.yaml#L53

**Reply by zskeep** [reply]
Thanks, type.The problem has been resolved.
Change the type to NodePort.

**Reply by DaCapoooo** [reply]
excuse me, how to Change the type to NodePort.

**Reply by CH3CHO** [reply]
Editing the Service resource would do.
